openmls/src/framing/plaintext.rs

```diff
@@ -30,145 +53,200 @@ use std::convert::TryFrom;
 /// opaque signature<0..2^16-1>;
 /// optional<MAC> confirmation_tag;
 /// optional<MAC> membership_tag;
-/// ```
 /// } MLSPlaintext;
+/// ```
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 pub struct MlsPlaintext {
-    pub(crate) group_id: GroupId,
-    pub(crate) epoch: GroupEpoch,
-    pub(crate) sender: Sender,
-    pub(crate) authenticated_data: Vec<u8>,
-    pub(crate) content_type: ContentType,
-    pub(crate) content: MlsPlaintextContentType,
-    pub(crate) signature: Signature,
-    pub(crate) confirmation_tag: Option<ConfirmationTag>,
-    pub(crate) membership_tag: Option<MembershipTag>,
+    group_id: GroupId,
+    epoch: GroupEpoch,
+    sender: Sender,
+    authenticated_data: Vec<u8>,
+    content_type: ContentType,
+    content: MlsPlaintextContentType,
+    signature: Signature,
+    confirmation_tag: Option<ConfirmationTag>,
+    membership_tag: Option<MembershipTag>,
+}
+
+// This block only has pub(super) getters.
+impl MlsPlaintext {
+    pub(super) fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    pub(super) fn membership_tag(&self) -> &Option<MembershipTag> {
+        &self.membership_tag
+    }
+
+    #[cfg(test)]
+    pub(super) fn signature_mut(&mut self) -> &mut Signature {
+        &mut self.signature
+    }
+
+    #[cfg(test)]
+    pub(super) fn unset_confirmation_tag(&mut self) {
+        self.confirmation_tag = None;
+    }
+
+    #[cfg(test)]
+    pub(super) fn unset_membership_tag(&mut self) {
+        self.membership_tag = None;
+    }
+
+    #[cfg(test)]
+    pub(super) fn set_content(&mut self, content: MlsPlaintextContentType) {
+        self.content = content;
+    }
+
+    #[cfg(test)]
+    pub(super) fn set_signature(&mut self, signature: Signature) {
+        self.signature = signature;
+    }
+
+    #[cfg(test)]
+    pub(super) fn set_membership_tag_test(&mut self, tag: MembershipTag) {
+        self.membership_tag = Some(tag);
+    }
 }
 
 impl MlsPlaintext {
-    /// This constructor builds a new `MlsPlaintext` from the parameters
-    /// provided. It is only used internally.
-    pub(crate) fn new_from_member(
+    /// Convenience function for creating an `MlsPlaintext`.
+    #[inline]
```

Comment:
This isn't forcing anything (you can't actually force the compiler to inline). It's just a hint that it would make sense to inline this because of the way it's used.